### ISSUE TITLE
false positives on redundant_objc_attribute

### ISSUE THREAD

**Issue by weakfl** [realm/SwiftLint#2499]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The `redundant_objc_attribute` rule yields false positives if a `private` function needs to be marked `@objc` in order to be used with a selector.

According to the [documentation](https://github.com/realm/SwiftLint/blob/master/Rules.md#redundant-@objc-attribute) this should be **not triggering**, but it is.

### Environment

* SwiftLint 0.29.1
* Homebrew

### Example

```swift
bar.addTarget(self, action: #selector(foo), for: .touchUpInside)
// This triggers a violation:
@objc private func foo() {
}
```

**Comment by marcelofabri** [comment]
Closed in #2501
